components/server/src/feature-flag/featureflag-controller.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { injectable, inject } from "inversify";
+import * as express from "express";
+import { Config } from "../config";
+import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import { User } from "@gitpod/gitpod-protocol";
+
+@injectable()
+export class FeatureFlagController {
+    @inject(Config) protected readonly config: Config;
+
+    get apiRouter(): express.Router {
+        const router = express.Router();
+        this.addSlowDatabaseFeatureFlagHandler(router);
+        return router;
+    }
+
+    protected addSlowDatabaseFeatureFlagHandler(router: express.Router) {
+        router.get("/slow-database", async (req, res) => {
+            if (!User.is(req.user)) {
+                res.sendStatus(400);
+                return;
+            }
+
+            const flagValue = await getExperimentsClientForBackend().getValueAsync("slow_database", false, {
```

Comment:
do we expect a delay on any hot path by introducing this?
I mean, that client-server talk may get slow, or break. For instance, what would a default timeout of 30s mean for the regular client connections? 

In doubt, let's race it agains a timeout promise of a ~2 seconds, and default to 'false' in this case. Also catching any other errors would be good, so the 200 OK contract is guaranteed.

Reply:
> Also catching any other errors would be good, so the 200 OK contract is guaranteed.

that'd be good, too.

Reply:
FYI, flags are fetched in the background and in practice, this shouldn't be a blocking call. It uses the await syntax because it would be a blocking if we used the "blocking fetch" mode of configcat, but because we update in the background it will by non-blocking here.